client/src/main/com/sinch/sdk/domains/conversation/adapters/AppService.java

```diff
@@ -0,0 +1,61 @@
+package com.sinch.sdk.domains.conversation.adapters;
+
+import com.sinch.sdk.core.exceptions.ApiException;
+import com.sinch.sdk.core.http.AuthManager;
+import com.sinch.sdk.core.http.HttpClient;
+import com.sinch.sdk.core.http.HttpMapper;
+import com.sinch.sdk.domains.conversation.adapters.api.v1.AppApi;
+import com.sinch.sdk.domains.conversation.adapters.converters.AppConverter;
+import com.sinch.sdk.domains.conversation.models.requests.AppRequestParameters;
+import com.sinch.sdk.domains.conversation.models.responses.App;
+import com.sinch.sdk.models.Configuration;
+import java.util.Collection;
+import java.util.Map;
+
+public class AppService implements com.sinch.sdk.domains.conversation.AppService {
+
+  private final Configuration configuration;
+  private final AppApi api;
+
+  public AppService(
+      Configuration configuration, HttpClient httpClient, Map<String, AuthManager> authManagers) {
+    this.configuration = configuration;
+    this.api =
+        new AppApi(
+            httpClient, configuration.getConversationServer(), authManagers, new HttpMapper());
+  }
+
+  protected AppApi getApi() {
+    return this.api;
+  }
+
+  public Collection<App> list() throws ApiException {
+    return AppConverter.convert(getApi().appListApps(configuration.getProjectId()));
+  }
+
+  public App get(String appId) {
+    return AppConverter.convert(getApi().appGetApp(configuration.getProjectId(), appId));
+  }
+
+  public void delete(String appId) {
+    getApi().appDeleteApp(configuration.getProjectId(), appId);
+  }
+
+  public App create(AppRequestParameters parameters) {
```

Comment:
Right.
By purpose, this class is same for both (name do not contains create of update reference) 
Waiting for feedback by API's owner team about required status:
 - display_name: is not really required because can be ignored if underlying API is called with. proper mask value
 - credentials array can be an empty array on create or update. So not really required.
When confirmation will be send, business interface could be updated accordingly and, if required, dedicated classes derivated from this one.

Note required fields are managed outside of dedicated business class but onto serialization at client level and based onto DTO description coming from OAS: so no need to write dedicated (and duplicated with DTO declaration) logic at business object level:
: https://github.com/sinch/sinch-sdk-java/blob/main/core/src/main/com/sinch/sdk/core/utils/databind/Mapper.java#L97-L100

But you pointed to a direction where there was a bug if credentials array was null onto creation 
 -> fixed

Reply:
Yes; my comment was ambiguous: display_name can be ignore at business level for server side (it mask is not set) but have to present onto payload: So yes it is required by protocol but not by business implementation 
Feedback I'm waiting for is about this required, but not.
If confirmation it his really required: SDK will need to choose:
- let display_name required for each request: the mean user will have to provide a value.
- use mask internally to avoid user seeing this field as required
My preference will be going to option 2